src/screens/AssignmentList/AssignmentListScreen.js

```diff
@@ -35,44 +36,117 @@ const styles = StyleSheet.create({
+            attachment,
+          })
+            .then(() => {
+              console.log('Successfully updated document!');
+            })
+            .catch((error) => {
+              console.log('Submission update failed');
+              console.log(error);
+            });
+        } else { // do this to add submission if it doesn't exist
+          Firestore().collection('submissions').add({
+            authorID: userID,
+            assignmentID: postID,
+            createdAt: firebase.firestore.FieldValue.serverTimestamp(),
+            updatedAt: firebase.firestore.FieldValue.serverTimestamp(),
+            body,
+            attachment,
+            hasReceivedFeedback: false,
+          }).then(() => {
+            console.log('Successfully inserted document!');
+          })
+            .catch((error) => {
+              console.log('Submission failed ):');
+              console.log(error);
+            });
+        }
+      });
+  });
+
+  useEffect(() => {
+    // Gets all the classrooms
+    Firestore().collection('classrooms').get()
       .then((snapshot) => {
-        const posts = snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
-        setPostsList(posts.map((post) => {
-          const date = post.createdAt.toDate();
-          return (
-            <View style={styles.container} key={post.id}>
-              <Post
-                key={post.id}
-                name={post.username}
-                title={post.title}
-                createdAt={date.toTimeString()}
-                date={date.toDateString()}
-                attachment={post.attachment}
-                body={post.body}
-              >
-                {post.body}
-              </Post>
-              <Button
-                styles={styles.container}
-                title="Comment on Post"
-                onPress={() => {
-                  navigation.navigate('NewComment', { ID: post.id });
-                }}
-              />
-            </View>
-          );
-        }));
+        const studentClassroomIDs = []; // array of all classrooms a student is in
+        const classrooms = snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
+        for (let i = 0; i < classrooms.length; i += 1) {
+          if (classrooms[i].studentIDs.includes(uid)) {
+            studentClassroomIDs.push(classrooms[i].id);
+          }
+        }
+        return studentClassroomIDs;
+      })
+      .then((studentClassroomIDs) => {
+        // Gets all the assignments
+        Firestore().collection('assignments').limit(6)
+          .orderBy('createdAt', 'desc')
+          .get()
+          .then((snapshot) => {
+            // This creates an object for each document
+            const posts = snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
+            setPostsList(posts.map((post) => {
+              const date = post.createdAt.toDate();
+              // Only returns assignments with classroomIDs in classroomIDsList
```

Comment:
Do assignments have classroomID arrays? Does the include() function work for strings?